digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/RoadLinkDAO.scala

```diff
@@ -346,14 +351,217 @@ class RoadLinkDAO {
+        Some(link.attributes.getOrElse("MODIFIED_BY", "").toString), link.attributes)
+    }
+  }
+
+  protected def getEnrichedRoadLinksWithFilter(bounds: BoundingRectangle, municipalities: Set[Int]): Seq[EnrichedRoadLinkFetched] = {
+    val bboxFilter = PostGISDatabase.boundingBoxFilter(bounds, "shape")
+    val municipalitiesFilter = if (municipalities.nonEmpty) s"AND municipalitycode IN (${municipalities.mkString(",")})" else ""
+    val constructionFilter = Seq(ConstructionType.Planned.value, ConstructionType.UnderConstruction.value).mkString(", ")
+    val query =
+      sql"""
+    WITH cte_kgv_roadlink AS (
+      SELECT
+        linkid, mtkid, mtkhereflip, municipalitycode, shape, adminclass, directiontype,
+        mtkclass, roadname_fi, roadname_se, roadnamesme, roadnamesmn, roadnamesms,
+        roadnumber, roadpartnumber, constructiontype, verticallevel, horizontalaccuracy,
+        verticalaccuracy, created_date, last_edited_date, from_left, to_left, from_right, to_right,
+        surfacetype, geometrylength
+      FROM kgv_roadlink kgv
+      WHERE
+        #$bboxFilter
+        #$municipalitiesFilter
+        AND kgv.expired_date IS NULL
+        AND kgv.constructiontype NOT IN (#$constructionFilter)
+        AND kgv.mtkclass NOT IN (12318, 12312) -- HardShoulder and WinterRoad
+    ),
+
+    cte_combined AS (
+      SELECT
+        cte.linkid, cte.mtkid, cte.mtkhereflip, cte.municipalitycode, cte.shape,
+        COALESCE(ac.administrative_class, cte.adminclass) AS adminclass,
+        cte.directiontype, td.traffic_direction, cte.mtkClass, fc.functional_class, lt.link_type,
+        cte.roadname_fi, cte.roadname_se, cte.roadnamesme, cte.roadnamesmn, cte.roadnamesms,
+        cte.roadnumber, cte.roadpartnumber, cte.constructiontype, cte.verticallevel,
+        cte.horizontalaccuracy, cte.verticalaccuracy, cte.created_date, cte.last_edited_date,
+        cte.from_left, cte.to_left, cte.from_right, cte.to_right, cte.surfacetype, cte.geometrylength,
+        ac.created_date AS ac_created_date, ac.created_by AS ac_created_by,
+        lt.modified_date AS lt_modified_date, lt.modified_by AS lt_modified_by,
+        fc.modified_date AS fc_modified_date, fc.modified_by AS fc_modified_by,
+        td.modified_date AS td_modified_date, td.modified_by AS td_modified_by
+      FROM cte_kgv_roadlink cte
+      LEFT JOIN administrative_class ac
+        ON cte.linkid = ac.link_id AND (ac.valid_to IS NULL OR ac.valid_to > current_timestamp)
+      LEFT JOIN link_type lt
+        ON cte.linkid = lt.link_id
+      LEFT JOIN functional_class fc
+        ON cte.linkid = fc.link_id
+      LEFT JOIN traffic_direction td
+        ON cte.linkid = td.link_id
+    )
+
+    SELECT
+      cte_combined.*,
+      ARRAY_AGG(ROW(rla.name, rla.value, COALESCE(rla.modified_date, rla.created_date), COALESCE(rla.modified_by, rla.created_by)))
+        FILTER (WHERE rla.name IS NOT NULL) AS attributes
```

Comment:
kansii nimetä private_road_information.
Nuo on yksityisteiden tiedot.

Reply:
attributes sana tuolla muuallakin koodia kansii nimetä paremmin.

Reply:
Liittyykö kaikki road_link_attributes sisältö pelkästään yksityisteihin? Ymmärsin että ainoastaan "PRIVATE_ROAD_ASSOCIATION" | "ADDITIONAL_INFO" on sellaisia, loput geneerisiä

Reply:
Näin ainakin minä olen ymmärtänyt. Voit kyl korjata jos oon väärässä.

Reply:
Janne vahvisti. Ilmeisesti taulu on tällä hetkellä vain yksityistietietojen käytössä, mutta nimetty ja toteutettu geneerisesti tulevaisuutta varten. Muutin termin koodissa taulun osalta private_road_infoksi, sen voi sitten nimetä uudestaan jos käyttötarkoitus uudistuu.